packages/suite/src/actions/wallet/moveLabelsForRbf.ts

```diff
@@ -0,0 +1,154 @@
+import { selectLabelingDataForAccount } from '../../reducers/suite/metadataReducer';
+import { findChainedTransactions, findTransactions } from '@suite-common/wallet-utils';
+import { Dispatch, GetState } from 'src/types/suite';
+import * as metadataActions from 'src/actions/suite/metadataActions';
+import { AccountLabels, AccountOutputLabels } from '@suite-common/metadata-types';
+import { AccountKey, WalletAccountTransaction } from '@suite-common/wallet-types';
+
+type DeleteAllOutputLabelsParams = {
+    labels: AccountLabels['outputLabels']['labels'];
+    dispatch: Dispatch;
+    accountKey: AccountKey;
+    txid: string;
+};
+
+const deleteAllOutputLabels = async ({
+    labels,
+    dispatch,
+    accountKey,
+    txid,
+}: DeleteAllOutputLabelsParams) => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const outputIndex of Object.keys(labels)) {
+        // eslint-disable-next-line no-await-in-loop
+        await dispatch(
+            metadataActions.addMetadata({
+                type: 'outputLabel',
+                entityKey: accountKey,
+                txid,
+                outputIndex: Number(outputIndex),
+                defaultValue: '',
+                value: '',
+            }),
+        );
+    }
+};
+
+type MoveLabelToNewTransactionParams = {
+    accountOutputLabels: AccountOutputLabels;
+    dispatch: Dispatch;
+    accountKey: AccountKey;
+    newTxid: string;
+};
+
+export const moveLabelToNewTransaction = async ({
+    accountOutputLabels,
+    dispatch,
+    accountKey,
+    newTxid,
+}: MoveLabelToNewTransactionParams) => {
+    // eslint-disable-next-line no-restricted-syntax,
+    for (const outputIndex of Object.keys(accountOutputLabels)) {
+        const value = accountOutputLabels[outputIndex];
+        // eslint-disable-next-line no-await-in-loop
+        await dispatch(
+            metadataActions.addMetadata({
+                type: 'outputLabel',
+                entityKey: accountKey,
+                txid: newTxid,
+                outputIndex: Number(outputIndex),
+                defaultValue: '',
+                value,
+            }),
+        );
+    }
+};
+
+type FindLabelsToBeMovedOrDeletedParams = {
+    prevTxid: string;
+    getState: GetState;
+};
+
+export type LabelsToBeMovedOrDeleted = Record<
+    AccountKey,
+    {
+        toBeMoved: WalletAccountTransaction;
+        toBeDeleted: WalletAccountTransaction[];
+    }
+>;
+
+export const findLabelsToBeMovedOrDeleted = ({
+    prevTxid,
+    getState,
```

Comment:
if written as a think, the get state is available without passing. I would rewrite 🙏

Reply:
:+1: